src/switch_core_media.c

```diff
@@ -2022,14 +2025,17 @@ SWITCH_DECLARE(void) switch_core_session_check_outgoing_crypto(switch_core_sessi
 	switch_log_printf(SWITCH_CHANNEL_SESSION_LOG(session), SWITCH_LOG_DEBUG, "CRYPTO: Checking outgoing crypto\n");
 
 	for (i = 0; smh->crypto_suite_order[i] != CRYPTO_INVALID; i++) {
+
+		switch_log_printf(SWITCH_CHANNEL_SESSION_LOG(session), SWITCH_LOG_DEBUG, "CRYPTO: Checking outgoing crypto: build it at index=%d\n", i + 1);
+
 		switch_core_media_build_crypto(session->media_handle,
-									   SWITCH_MEDIA_TYPE_AUDIO, SWITCH_NO_CRYPTO_TAG, smh->crypto_suite_order[i], SWITCH_RTP_CRYPTO_SEND, 0, 0);
+									   SWITCH_MEDIA_TYPE_AUDIO, i + 1, smh->crypto_suite_order[i], SWITCH_RTP_CRYPTO_SEND, 0, 0);
```

Comment:
I was checking inside switch_core_media_build_crypto, so I noticed that if index is set to `SWITCH_NO_CRYPTO_TAG` it will performed this logic:
```
if (index == SWITCH_NO_CRYPTO_TAG) index = ctype + 1;
```
This seems similar logic when we set the index to `i+1`

Reply:
Yes, this is what this fix changes.

Reply:
It calls switch_core_media_build_crypto with index instead of SWITCH_NO_CRYPTO_TAG (which is defined to -1), to make crypto built from 1 to n.

Reply:
This way if there is single crypto to add it will get built with index 1.